torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -121,5 +138,26 @@ c10::intrusive_ptr<ProcessGroup::Work> allgather(
       output_tensors, input_tensors, process_group, opts.timeout.count());
 }
 
+c10::intrusive_ptr<ProcessGroup::Work> reduce_scatter(
+    const c10::intrusive_ptr<ProcessGroup>& process_group,
+    const std::vector<at::Tensor>& output_tensors,
+    const std::vector<std::vector<at::Tensor>>& input_tensors,
+    const ReduceScatterOptions& opts) {
+  static auto op = c10::Dispatcher::singleton()
+                       .findSchemaOrThrow("c10d::reduce_scatter_", "")
+                       .typed<c10::intrusive_ptr<::c10d::ProcessGroup::Work>(
+                           const std::vector<at::Tensor>&,
+                           const std::vector<std::vector<at::Tensor>>&,
+                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
+                           int64_t,
+                           int64_t)>();
```

Comment:
Would I be correct that the overhead of these two ops are negligible?

Reply:
It should be negligible. And then we cache it for the consecutive calls to minimize the overhead too.